vars/commonPipelineEnvironment.groovy

```diff
@@ -46,6 +46,11 @@ class commonPipelineEnvironment implements Serializable {
     String mtarFilePath = ""
 
     String abapRepositoryNames
+    String abapAddonProduct
```

Comment:
what about adding the abap properties in a similar way like the container properties and use a Map?

Reply:
As far as I understood, the cobra framework is not able to handle the required data structures ([]map[string]string). Therefore it would probably be best to store it as a string.
Please correct me if I'm wrong - I would rather use maps 😉.

Reply:
cobra can handle a list of strings but nothing more special ...

This is the groovy part though and here you have all flexibility.
With `readFromDisk` and `writeFromDisk` you can do the transformations as also done for containers.